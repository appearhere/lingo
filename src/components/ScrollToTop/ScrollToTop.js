// eslint-disable-next-line
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';

class ScrollToTop extends Component {
  static propTypes = {
    location: PropTypes.shape({ }),
    children: PropTypes.node,
  };

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      window.scrollTo(0, 0);
    }
  }

  render() {
    return this.props.children;
  }
}

export default withRouter(ScrollToTop);
