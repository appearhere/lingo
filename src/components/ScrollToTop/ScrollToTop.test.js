import React from 'react';
import { render } from 'react-dom';

import ScrollToTop from './ScrollToTop';

it('renders without crashing', () => {
  const div = document.createElement('div');
  render(<ScrollToTop />, div);
});
