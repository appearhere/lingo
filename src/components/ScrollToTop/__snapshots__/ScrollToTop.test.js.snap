// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop component renders correctly 1`] = `<div />`;
